### PR TITLE
chore(ci): split build and tests jobs in `abi_wasm.yml`

### PR DESCRIPTION
**Pull request by TomAFrench**
# Description

## Summary\*

This separates the building and testing of `noirc_abi_wasm` which, among other things, allows easier rerunning in the case of a flakey test failure.

## Documentation

- [ ] This PR requires documentation updates when merged.

  

  - [ ] I will submit a noir-lang/docs PR.

  

  - [ ] I will request for and support Dev Rel's help in documenting this PR.

  
  

## Additional Context

# PR Checklist\*

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: .github/workflows/abi_wasm.yml [hunk 1]
```diff
@@ -72,6 +72,20 @@ jobs:
...
+    needs: [build]
+  
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v3
+
+      - name: Download wasm package artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: noir_wasm
```
Comment (kevaundray) [inline .github/workflows/abi_wasm.yml:86]:
```suggestion
          name: noirc_abi_wasm
```

**Comment by TomAFrench** [comment]
I've found the underlying cause for this issue.

**Comment by TomAFrench** [comment]
Oof, requested review too soon.

File: tooling/noirc_abi_wasm/package.json [hunk 2]
```diff
@@ -21,7 +21,6 @@
     "type": "git",
     "url": "https://github.com/noir-lang/noir.git"
   },
-  "packageManager": "yarn@3.5.1",
```
Comment (kevaundray) [inline tooling/noirc_abi_wasm/package.json:24]: why is this removed?

Comment (TomAFrench) [reply tooling/noirc_abi_wasm/package.json:24]: This is defined at the workspace root now. These are leftovers from when we didn't have a workspace.

Comment (kevaundray) [reply tooling/noirc_abi_wasm/package.json:24]: Ah we have this in two other crates I think -- I'll push another PR to remove those
